### PR TITLE
NBC-32 # Evergreen - Overwrite Files Prompt

### PR DESCRIPTION
**Pull request by mymattcarroll**
### Changed
- NBC-32: `bm buildbot evergreen` cmd with no `--force` flag functionality 
  - Will now prompt for confirmation to overwrite existing files instead of throwing an error
  - `--force` flag functionality has not changed, it now by passes the confirmation prompt

### CODE REVIEW

**Comment by jokeyrhyme** [comment]
@mymattcarroll there are merge conflicts now, probably from a recent PR

File: package.json [hunk 1]
```diff
@@ -33,9 +34,10 @@
     "@jokeyrhyme/pify-fs": "1.0.1",
     "archiver": "1.1.0",
     "aws-sdk": "^2.5.3",
+    "chalk": "^1.1.3",
     "elegant-spinner": "^1.0.1",
     "execa": "^0.4.0",
-    "inquirer": "^1.0.3",
+    "inquirer": "^1.2.1",
```
Comment (jokeyrhyme) [inline package.json]: It is theoretically easier to reproduce and thus diagnose reported issues if production dependencies are locked (not a ~ or ^ range). This is a CLI, not a library, so I think locking the versions is more appropriate.

File: test/evergreen.js [hunk 2]
```diff
@@ -30,9 +74,113 @@ test('findPlatforms: none', (t) => {
...
+  return evergreen.getCurrentTenant()
+    .then((tenant) => t.is(tenant, TENANT_NAME));
+});
+
+test('getCurrentTenant() should reject if no current tenant is found', (t) => {
+  const evergreen = t.context.getTestSubject({
+    '@blinkmobile/bm-identity': function BlinkMobileIdentity () {
+      this.getTenants = () => {
+        return Promise.resolve();
+      };
```
Comment (jokeyrhyme) [inline test/evergreen.js]: Shorter (although not necessarily clearer): `this.getTenants = () => Promise.resolve()`?

File: test/evergreen.js [hunk 3]
```diff
@@ -3,10 +3,52 @@
...
+      'lodash.memoize': (fn) => fn,
+      'inquirer': inquirerMock((questions) => {
+        return Promise.resolve({
+          confirmation: true
+        });
+      }),
+      '@blinkmobile/bm-identity': function BlinkMobileIdentity () {
+        this.assumeAWSRole = () => {
+          return Promise.resolve();
+        };
```
Comment (jokeyrhyme) [inline test/evergreen.js]: Shorter (although not necessarily clearer): `this.assumeAWSRole = () => Promise.resolve()`?

File: test/evergreen.js [hunk 4]
```diff
@@ -3,10 +3,52 @@
...
+        this.getTenants = () => {
+          return Promise.resolve({
+            current: TENANT_NAME
+          });
+        };
```
Comment (jokeyrhyme) [inline test/evergreen.js]: Shorter (although not necessarily clearer):

``` js
this.getTenants = () => Promise.resolve({
  current: TENANT_NAME
})
```

?

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,14 @@
 
 ## Next
```
Comment (jokeyrhyme) [inline CHANGELOG.md]: If we're sticking to http://keepachangelog.com/ , then this should probably be `## Unreleased` or `## [Unreleased]`

File: lib/help.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ Evergreen update:
   evergreen <project_path>      => generate evergreen update ZIPs
                                      omit <project_path> to use working dir
     --force                     => overwrite any existing update ZIPs
+                                     omit to be prompted for confirmation
```
Comment (jokeyrhyme) [inline lib/help.js]: Shorter to state `-- force => overwrite existing files without confirmation` ? Just assume that the default behaviour is the opposite (with confirmation)?

**Review by jokeyrhyme — CHANGES_REQUESTED** [review]

**Comment by jokeyrhyme** [comment]
I clicked "Request Changes" in the GitHub Review feature thing just to see what it does. Feel free to knock me back. :)

File: CHANGELOG.md (hunk 5 above)
Comment (mymattcarroll) [reply CHANGELOG.md]: Cool, thanks. Will change to `Unreleased`

File: lib/help.js (hunk 6 above)
Comment (mymattcarroll) [reply lib/help.js]: Agreed.

File: package.json (hunk 1 above)
Comment (mymattcarroll) [reply package.json]: That sounds like a good plan to me, CLI === looked dependencies, libraries === ranged dependencies

**Review by jokeyrhyme — APPROVED** [review]

**Pull request merged**